### PR TITLE
Allow External ID Resolution via local MCF files

### PR DESCRIPTION
**Pull request by enjoythecode**
### Old Behavior

All external ID (namespaces such as `isoCode`) resolution happens against the recon API of Data Commons. During a dataset import where new external IDs are needed for resolution, those IDs must be pushed to the recon API, re-deployed... which slows down new imports.

### New Behavior

Input Instance MCFs to the import tool are added to the local graph of `ExternalIdResolver` if they are an externally resolvable type (as determined by `ExternalIdResolver.isResolvableType()`). When resolution calls are made for these IDs, they are resolved locally without any API calls.

### Testing

I have used existing tests within `ExternalIdResolver` (which was written for API calls) by refactoring the common test suite to a function and handling only the input (API calls for remote, loading MCF nodes in-memory for local graph).

### CODE REVIEW

**Comment by enjoythecode** [comment]
TODO(snny): add an external ID test by loading from a local MCF, rather than in-memory nodes.

File: tool/src/test/resources/org/datacommons/tool/genmcf/localidresolution/input/schema.mcf [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Node: dcid:CumulativeCount_MedicalTest_ConditionCOVID_19_Positive
+typeOf: dcs:StatisticalVariable
```
Comment (pradh) [inline tool/src/test/resources/org/datacommons/tool/genmcf/localidresolution/input/schema.mcf:2]: Don't need this?

**Review by pradh — APPROVED** [review]
Nicely done!

**Comment by pradh** [comment]
Looks like this is failing test?

**Comment by pradh** [comment]
> Looks like this is failing test?

[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.545 s <<< FAILURE! - in org.datacommons.tool.GenMcfTest
[ERROR] GenMcfTest(org.datacommons.tool.GenMcfTest)  Time elapsed: 0.542 s  <<< FAILURE!
org.junit.ComparisonFailure: The text content of the actual HTML file differed from expected expected:<...LEMENT WITHOUT TEXT>[<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>CumulativeCount_MedicalTest_ConditionCOVID_19_Positive<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>Existence Checks EnabledyesResolution ModeRESOLUTION_MODE_FULLNum Threads1Stat Checks EnabledyesSample Places Entered<ELEMENT WITHOUT TEXT>Counter NameNum Occurences<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>11<ELEMENT WITHOUT TEXT>33<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1StatVarNum PlacesNum ObservationsNum Observation DatesMin DateMax DateMeasurement MethodsUnitsScaling FactorsObservation Periods<ELEMENT WITHOUT TEXT>1112020-01-012020-01-01<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_INFO<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_ERROR<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple]<ELEMENT WITHOUT TEX...> but was:<...LEMENT WITHOUT TEXT>[Open this place (pseudoPlaceSimple) in Data Commons browser.<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple (pseudoPlaceSimple)<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>CumulativeCount_MedicalTest_ConditionCOVID_19_Positive<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>Existence Checks EnabledyesResolution ModeRESOLUTION_MODE_FULLNum Threads1Stat Checks EnabledyesSample Places Entered<ELEMENT WITHOUT TEXT>Counter NameNum Occurences<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>11<ELEMENT WITHOUT TEXT>33<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1StatVarNum PlacesNum ObservationsNum Observation DatesMin DateMax DateMeasurement MethodsUnitsScaling FactorsObservation Periods<ELEMENT WITHOUT TEXT>1112020-01-012020-01-01<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_INFO<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_ERROR<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple (pseudoPlaceSimple)]<ELEMENT WITHOUT TEX...>
	at org.datacommons.tool.GenMcfTest.GenMcfTest(GenMcfTest.java:118)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   GenMcfTest.GenMcfTest:118 The text content of the actual HTML file differed from expected expected:<...LEMENT WITHOUT TEXT>[<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>CumulativeCount_MedicalTest_ConditionCOVID_19_Positive<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>Existence Checks EnabledyesResolution ModeRESOLUTION_MODE_FULLNum Threads1Stat Checks EnabledyesSample Places Entered<ELEMENT WITHOUT TEXT>Counter NameNum Occurences<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>11<ELEMENT WITHOUT TEXT>33<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1StatVarNum PlacesNum ObservationsNum Observation DatesMin DateMax DateMeasurement MethodsUnitsScaling FactorsObservation Periods<ELEMENT WITHOUT TEXT>1112020-01-012020-01-01<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_INFO<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_ERROR<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple]<ELEMENT WITHOUT TEX...> but was:<...LEMENT WITHOUT TEXT>[Open this place (pseudoPlaceSimple) in Data Commons browser.<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple (pseudoPlaceSimple)<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>CumulativeCount_MedicalTest_ConditionCOVID_19_Positive<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>Existence Checks EnabledyesResolution ModeRESOLUTION_MODE_FULLNum Threads1Stat Checks EnabledyesSample Places Entered<ELEMENT WITHOUT TEXT>Counter NameNum Occurences<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>11<ELEMENT WITHOUT TEXT>33<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT>2<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1<ELEMENT WITHOUT TEXT>1StatVarNum PlacesNum ObservationsNum Observation DatesMin DateMax DateMeasurement MethodsUnitsScaling FactorsObservation Periods<ELEMENT WITHOUT TEXT>1112020-01-012020-01-01<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_INFO<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>LEVEL_ERROR<ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT><ELEMENT WITHOUT TEXT>pseudoPlaceSimple (pseudoPlaceSimple)]<ELEMENT WITHOUT TEX...>
[INFO] 
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0

**Pull request merged**